eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java

```diff
@@ -0,0 +1,21 @@
+package io.eventuate.common.quarkus.id;
+
+import io.eventuate.common.id.ApplicationIdGenerator;
+import io.eventuate.common.id.IdGenerator;
+import io.eventuate.common.id.DatabaseIdGenerator;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import java.util.Optional;
+
+@ApplicationScoped
+public class IdGeneratorConfiguration {
+  @Produces
+  public IdGenerator idGenerator(@ConfigProperty(name = "eventuate.outbox.id") Optional<Long> id) {
+    return id
```

Comment:
Is this code that is duplicated elsewhere?

Reply:
Do these four lines of code also occur in eventuate-common?

```
id
            .map(DatabaseIdGenerator::new)
            .map(generator -> (IdGenerator)generator)
            .orElseGet(ApplicationIdGenerator::new);
```

Reply:
No, it looks different for micronaut and spring.
spring: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-spring-id/src/main/java/io/eventuate/common/spring/id/IdGeneratorConfiguration.java
micronaut: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-id/src/main/java/io/eventuate/common/micronaut/id/IdGeneratorFactory.java